.github/workflows/Automation Master Legacy Workflow.yml

```diff
@@ -279,40 +279,118 @@ jobs:
+          run: |
+            $SOURCE_NAME="PrivateGitHubNuGets"
+            $SOURCE_URL="https://nuget.pkg.github.com/SkylineCommunications/index.json"
+            
+            # Check if the source exists. If it does, update it.
+            if (nuget sources list | Select-String -Pattern $SOURCE_NAME) {
+              Write-Host "Updating existing source $SOURCE_NAME."
+              nuget sources update -name $SOURCE_NAME -source $SOURCE_URL -username USERNAME -password ${{ secrets.GITHUB_TOKEN }} -StorePasswordInClearText
+            } else {
+              Write-Host "Adding new source $SOURCE_NAME."
+              nuget sources add -source $SOURCE_URL -name $SOURCE_NAME -username USERNAME -password ${{ secrets.GITHUB_TOKEN }} -StorePasswordInClearText
+            }
+          shell: pwsh
+
+        - name: Enable Skyline Azure Cloud NuGet Registry
+          env:
+            AZURE_TOKEN_EXISTS: ${{ secrets.azureToken }}
+          if: env.AZURE_TOKEN_EXISTS != null && inputs.owner == 'SkylineCommunications'
+          run: |
+            $SOURCE_NAME="CloudNuGets"
+            $SOURCE_URL="https://pkgs.dev.azure.com/skyline-cloud/Cloud_NuGets/_packaging/CloudNuGet/nuget/v3/index.json"
+            
+            # Check if the source exists. If it does, update it.
+            if (nuget sources list | Select-String -Pattern $SOURCE_NAME) {
+              Write-Host "Updating existing source $SOURCE_NAME."
+              nuget sources update -name $SOURCE_NAME -source $SOURCE_URL -username az -password ${{ secrets.azureToken }} -StorePasswordInClearText
+            } else {
+              Write-Host "Adding new source $SOURCE_NAME."
+              nuget sources add -source $SOURCE_URL -name $SOURCE_NAME -username az -password ${{ secrets.azureToken }} -StorePasswordInClearText
+            }
+          shell: pwsh
+
+        - name: Enable Skyline Azure Private NuGet Registry
+          env:
+            AZURE_TOKEN_EXISTS: ${{ secrets.azureToken }}
+          if: env.AZURE_TOKEN_EXISTS != null && inputs.owner == 'SkylineCommunications'
+          run: |
+            $SOURCE_NAME="PrivateAzureNuGets"
+            $SOURCE_URL="https://pkgs.dev.azure.com/skyline-cloud/_packaging/skyline-private-nugets/nuget/v3/index.json"
+            
+            # Check if the source exists. If it does, update it.
+            if (nuget sources list | Select-String -Pattern $SOURCE_NAME) {
+              Write-Host "Updating existing source $SOURCE_NAME."
+              nuget sources update source -name $SOURCE_NAME -source $SOURCE_URL -username az -password ${{ secrets.azureToken }} -StorePasswordInClearText
+            } else {
+              Write-Host "Adding new source $SOURCE_NAME."
+              nuget sources add -source $SOURCE_URL -name $SOURCE_NAME -username az -password ${{ secrets.azureToken }} -StorePasswordInClearText
+            }
+          shell: pwsh
+  
+        - name: Restore NuGet
+          run: nuget restore
```

Comment:
NuGet restore doesn't have the solution filepath as argument as how it was before.